diff --git a/node_modules/typescript/lib/tsc.js b/node_modules/typescript/lib/tsc.js
index 361e447..11c9ca9 100644
--- a/node_modules/typescript/lib/tsc.js
+++ b/node_modules/typescript/lib/tsc.js
@@ -21961,6 +21961,19 @@ var ts;
             description: ts.Diagnostics.Type_declaration_files_to_be_included_in_compilation,
             transpileOptionValue: undefined
         },
+        {
+            name: "excludeTypes",
+            type: "list",
+            element: {
+                name: "excludeTypes",
+                type: "string"
+            },
+            affectsModuleResolution: true,
+            showInSimplifiedHelpView: true,
+            category: ts.Diagnostics.Module_Resolution_Options,
+            description: ts.Diagnostics.Type_declaration_files_to_be_included_in_compilation,
+            transpileOptionValue: undefined
+        },
         {
             name: "allowSyntheticDefaultImports",
             type: "boolean",
@@ -24107,6 +24120,9 @@ var ts;
                     if (host.directoryExists(root)) {
                         for (var _a = 0, _b = host.getDirectories(root); _a < _b.length; _a++) {
                             var typeDirectivePath = _b[_a];
+                            if (options.excludeTypes && options.excludeTypes.includes(typeDirectivePath)) {
+                                continue;
+                            }
                             var normalized = ts.normalizePath(typeDirectivePath);
                             var packageJsonPath = ts.combinePaths(root, normalized, "package.json");
                             var isNotNeededPackage = host.fileExists(packageJsonPath) && ts.readJson(packageJsonPath, host).typings === null;
